cmd/fsi.go

```diff
@@ -108,10 +109,16 @@ func (o *FSIOptions) Unlink() error {
 		printRefSelect(o.ErrOut, o.Refs)
 
 		p := &lib.LinkParams{
-			Dir: o.Refs.Dir(),
 			Ref: ref,
 		}
 
+		dsRef := reporef.DatasetRef{}
+		if err := o.FSIMethods.FSIDatasetForRef(&ref, &dsRef); err != nil {
```

Comment:
Note that the implementation of `FSIDatasetForRef` calls `CanonicalizeDatasetRef`, which is how it's getting the `FSIPath`. I'm actually not a fan of `FSIDatasetForRef`, it's not being called anywhere in the codebase, and I'm of the opinion that it should be deleted since it's not particularly doing any work.